src/Fable.Cli/Main.fs

```diff
@@ -144,11 +144,56 @@ module private Util =
             if fileExt.EndsWith(".ts") then Path.replaceExtension ".js" fileExt else fileExt
         let sourceDir = Path.GetDirectoryName(sourcePath)
         let targetDir = Path.GetDirectoryName(targetPath)
-        let stream = new IO.StreamWriter(targetPath)
+        let memoryStream = new IO.MemoryStream()
+        let stream = new IO.StreamWriter(memoryStream)
         let mapGenerator = lazy (SourceMapSharp.SourceMapGenerator(?sourceRoot = cliArgs.SourceMapsRoot))
+        let WriteStreamToFileAsync(stream : IO.Stream, filePath : string) = async {
+            stream.Seek(0, IO.SeekOrigin.Begin) |> ignore
+            use fileStream = new IO.StreamWriter(filePath)
+            do! stream.CopyToAsync(fileStream.BaseStream) |> Async.AwaitTask
+            do! fileStream.FlushAsync() |> Async.AwaitTask
+        }
+        let IsMemoryStreamEqualToFileAsync() = async {
+            memoryStream.Seek(0, IO.SeekOrigin.Begin) |> ignore
+            use fileStream = IO.File.OpenRead(targetPath)
+
+            let bufferSize = 1024
+
+            let readBytesAsync (stream: IO.Stream) size = async {
+                let buffer = Array.zeroCreate size
+                let rec nextBytesAsync() = async {
+                    let! count = stream.AsyncRead(buffer, 0, size)
+                    if count = 0 then return None
+                    else 
+                        let res = 
+                            if count = size then buffer
+                            else buffer |> Seq.take count |> Array.ofSeq
+                        return Some(res)
+                }
+                return! nextBytesAsync()
+            }
+
+            let rec compareBytesAsync bytes1 bytes2 = async {
+                match bytes1, bytes2 with
+                | Some(b1), Some(b2) when b1 <> b2 -> return false
+                | Some(_), Some(_) -> 
+                    let! nextBytes1 = readBytesAsync memoryStream bufferSize
+                    let! nextBytes2 = readBytesAsync fileStream bufferSize
+                    return! compareBytesAsync nextBytes1 nextBytes2
+                | None, None -> return true
+                | _ -> return false
+            }
+
+            let! memoryBytes = readBytesAsync memoryStream bufferSize
+            let! fileBytes = readBytesAsync fileStream bufferSize
+            return! compareBytesAsync memoryBytes fileBytes
```

Comment:
You could first compare `memoryBytes.Lenght` and the file size before starting to read the latter from disk and comparing byte by byte.